File: app/src/main/java/zebrostudio/wallr100/ui/base_fragment/BaseFragment.java
package zebrostudio.wallr100.ui.base_fragment;


import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import javax.inject.Inject;

import dagger.android.support.DaggerFragment;
import zebrostudio.wallr100.R;

public class BaseFragment extends DaggerFragment implements
        BaseFragmentContract.BaseFragmentView {

    @Inject
    BaseFragmentPresenterImpl mBasePresenter;

    @Inject
    public BaseFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_base, container, false);
    }

    @Override
    public void onResume() {
        super.onResume();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        mBasePresenter.bindView(this);
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mBasePresenter.unbindView();
    }
}

File: app/src/main/java/zebrostudio/wallr100/ui/basefragment/BaseFragment.kt
package zebrostudio.wallr100.ui.basefragment

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import com.zebrostudio.wallrcustoms.customtextview.WallrCustomTextView
import dagger.android.support.AndroidSupportInjection
import zebrostudio.wallr100.R
import zebrostudio.wallr100.utils.setMenuItemColorRed
import zebrostudio.wallr100.utils.setMenuItemColorWhite

abstract class BaseFragment : Fragment() {

  internal lateinit var fragmentTag: String

  private val menuItemIdList: List<Int> = listOf(
      R.string.guillotine_explore_title,
      R.string.guillotine_top_picks_title,
      R.string.guillotine_categories_title,
      R.string.guillotine_minimal_title,
      R.string.guillotine_collection_title
  )

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    AndroidSupportInjection.inject(this)
  }

  @SuppressLint("ResourceType")
  override fun onResume() {
    super.onResume()
    activity?.findViewById<WallrCustomTextView>(R.id.toolbarTitle)?.text = fragmentTag

    highlightCurrentMenuItem()
    showAppropriateToolbarMenuIcon()

  }

  private fun highlightCurrentMenuItem() {
    for (menuItem in menuItemIdList) {
      if (getString(menuItem) == fragmentTag) {
        activity?.findViewById<LinearLayout>(menuItem)?.setMenuItemColorRed(this.context!!)
      } else {
        activity?.findViewById<LinearLayout>(menuItem)?.setMenuItemColorWhite(this.context!!)
      }
    }
  }

  private fun showAppropriateToolbarMenuIcon() {
    activity?.findViewById<ImageView>(R.id.toolbarMultiSelectIcon)?.visibility = View.GONE
    activity?.findViewById<ImageView>(R.id.toolbarSearchIcon)?.visibility = View.GONE
    when (fragmentTag) {
      getString(R.string.guillotine_minimal_title) ->
        activity?.findViewById<ImageView>(R.id.toolbarMultiSelectIcon)?.visibility = View.VISIBLE
      getString(R.string.guillotine_collection_title) -> {  // Do nothing
      }
      else -> activity?.findViewById<ImageView>(R.id.toolbarSearchIcon)?.visibility = View.VISIBLE
    }
  }

}